lib/puppet/provider/package/puppet_gem.rb

```diff
@@ -5,10 +5,7 @@
   has_feature :versionable, :install_options, :uninstall_options
 
   if Puppet::Util::Platform.windows?
-    # On windows, we put our ruby ahead of anything that already
-    # existed on the system PATH. This means that we do not need to
-    # sort out the absolute path.
-    commands :gemcmd => "gem"
+    commands :gemcmd => File.join(Puppet::Util.get_env('PUPPET_DIR').to_s, 'bin', 'gem.bat')
```

Comment:
puppet_gem provider from puppet gem will not know about `PUPPET_DIR` and it will say that puppet_gem provider cannot is not functional even after installing Puppet Agent AIO.

On linux, this provider will install in Puppet's Ruby if Puppet Agent AIO is present because the gem path is known `/opt/puppetlabs/puppet/bin/gem`